src/index.tsx

```diff
@@ -36,13 +37,13 @@ root.render(
       <Notifications position="top-right" zIndex={1000} />
       <RepoContext.Provider value={repo}>
         <BrowserRouter>
-          <ModalsProvider>
-            <DatesProvider settings={{ locale: "it" }}>
-              <DocProvider>
+          <DocProvider>
+            <ModalsProvider modals={modals}>
+              <DatesProvider settings={{ locale: "it" }}>
```

Comment:
sure that DatesProvider should be inside ModalsProvider? because in this way modalsProvider loads the modal components outside it